Makefile

```diff
@@ -0,0 +1,158 @@
+VIRTUALENV=~/.venvs/pixua
+
+.PHONY: setup
+setup: requirements install
+
+.PHONY: requirements
+requirements: requirements_py requirements_js
+	@echo "Installed requirements"
+
+.PHONY: install
+install: requirements install_py install_js
+	@echo "Installed dependencies"
+
+.PHONY: requirements_py
+requirements_py: create_virtualenv
+	@echo "Upgrading pip and setuptools"
+	@source $(VIRTUALENV)/bin/activate;\
+	pip install --upgrade pip setuptools
+
+.PHONY: install_py
+install_py: service/requirements.txt service/test_requirements.txt
+	@echo "Installing python packages"
+	@source $(VIRTUALENV)/bin/activate;\
+	cd service;\
+	pip install --exists-action s -r requirements.txt -r test_requirements.txt
+
+.PHONY: requirements_js
+requirements_js:
+	@echo "Installing javascript npm and bower dependencies"
+	@cd web-ui;\
+	npm install;\
+	node_modules/.bin/bower install
+
+.PHONY: install_js
+install_js:
+	@echo "Building front-end and static files"
+	@cd web-ui;\
+	npm run build
+
+.PHONY: create_virtualenv
+create_virtualenv: ensure_virtualenv_installed
+	@if [ ! -e $(VIRTUALENV)/bin/activate ]; then\
+		echo "Pixelated virtualenv doesn't exist, creating now";\
+		virtualenv --python=python2 $(VIRTUALENV);\
+	else\
+    echo "Pixelated virtualenv already exists, moving on";\
+	fi
+
+.PHONY: test
+test: test_py test_js coverage
+
+.PHONY: test_py
+test_py: clean requirements install coverage linters_py unit_tests_py integration_tests_py
+
+.PHONY: test_js
+test_js: clean requirements_js install_js linters_js unit_tests_js
```

Comment:
I don't think we need to call `clean` here, since it'll run twice. Maybe we can specify `clean-js` and `clean-py` for the two test tasks. I know it's super quick, but we can take any performance gain + clearer code.